# HG changeset patch
# User Rik <rik@octave.org>
# Date 1381950968 25200
#      Wed Oct 16 12:16:08 2013 -0700
# Node ID 171e958d672ed9fac632e3926aa11b58f0b6620a
# Parent  243116aa0e44e25ab462ff6cb8f2f137b4a19f3f
line-continue.tst: Add tests for '\' used in strings.

* test/line-continue.tst: Add tests for '\' used in strings.
Add note about removing tests for '\' as a line continuation because
it is now deprecated.

diff --git a/test/line-continue.tst b/test/line-continue.tst
--- a/test/line-continue.tst
+++ b/test/line-continue.tst
@@ -21,16 +21,19 @@
 %! a = 1;
 %! b = 2;
 
 %!test
 %! y = [a... # comments here ok
 %! b];
 %! assert (y, x);
 
+## FIXME: Next 3 tests use '\' continuation outside of a double-quoted string
+##        This behavior is deprecated and will be removed at some point.
+##        When it does these 
 %!test
 %! x = [1;2];
 %! y = [a... # comments here ok
 %! ;\
 %!
 %! b];
 %! assert (y, x);
 
@@ -46,16 +49,23 @@
 %!                y,  \# but they should work too.
 %!                z)
 %!
 %!  y = 1;
 %!endfunction
 %!
 %!assert (f (), 1)
 
+# String continuation using '\'
+%!assert (["abc\
+%! def"], "abc def")
+
+%!assert (['abc\
+%! def'], "abc def")
+
 %!test
 %!assert (1 == 1
 %! && 2 == 2
 %! || 3 == 5);
 
 %!test
 %! x = [1, ...
 %!
